### PR TITLE
dev-util/git-delta: add 0.17.0

### PR DESCRIPTION
**Pull request by imwints**
---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: dev-util/git-delta/git-delta-0.17.0.ebuild
Comment (thesamesam) [inline dev-util/git-delta/git-delta-0.17.0.ebuild:222]: I didn't bump it yet precisely because of this. It didn't feel like an improvement to force a downgrade of libgit2.

Comment (imwints) [reply dev-util/git-delta/git-delta-0.17.0.ebuild:222]: I could remove the subslot and the export line but then it would use the bundled libgit2 which is version 1.7.2 as well

File: dev-util/git-delta/git-delta-0.17.0.ebuild
Comment (eli-schwartz) [inline dev-util/git-delta/git-delta-0.17.0.ebuild:231]:
```
git --no-pager log -p -1 -C05 FETCH_HEAD -- 'dev-util/git-delta/*.ebuild'
```

Shows this diff:

```diff
 RDEPEND="
 	${DEPEND}
 	!app-text/delta
 "
 
-QA_FLAGS_IGNORED="usr/bin/delta"
+QA_FLAGS_INGORED="usr/bin/delta"
 ```

Seems like a bit of an oops?

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: dev-util/git-delta/git-delta-0.17.0.ebuild
Comment (eli-schwartz) [reply dev-util/git-delta/git-delta-0.17.0.ebuild:222]: https://github.com/rust-lang/git2-rs/pull/1032

Of course it is not yet released... so no one releasing new versions of their own projects can update libgit2 either.

Comment (imwints) [reply dev-util/git-delta/git-delta-0.17.0.ebuild:222]: I get that.

What I was trying to do is to use the system library as recommended in the ebuild guide for rust. However doing this requires both the `export LIBGIT2_NO_VENDOR=1` and `libgit2-1.7.x` to be present so I pinned the subslot. git-delta-0.16.5 on my system was not linked against libgit since I had some 1.8 version installed.

I was wondering if this is even worth doing or if we could just use the vendored libgit and remove the dependency in the ebuild so it could be independently upgraded if this blocks the PR. So basically keep it as in the old ebuild.

**Comment by thesamesam** [comment]
On reflection, you're right. Thanks!

**Comment by imwints** [comment]
@thesamesam 

This is not the only package that has this issue. Should I change them too or should this be up to the individual maintainer?

**Pull request closed without merging**
